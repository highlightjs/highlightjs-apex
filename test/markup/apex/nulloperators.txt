trigger OppLineSyncTrigger on OpportunityLineItem(after insert, after update) { //NOPMD
	if (Quote_Custom_Sync_Config__mdt.getInstance('Use_Included_Triggers')?.Enabled__c ?? false) {
		new QuoteSyncHandler().syncOppLines(Trigger.new, Trigger.oldMap);
	}

	Integer notNullReturnValue = (anInteger != null) ? anInteger : 100;

	Integer notNullReturnValue = anInteger ?? 100;
}