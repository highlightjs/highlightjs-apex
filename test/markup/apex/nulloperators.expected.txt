<span class="hljs-keyword">trigger</span> <span class="hljs-title class_">OppLineSyncTrigger</span> <span class="hljs-operator">on</span> <span class="hljs-type">OpportunityLineItem</span>(<span class="hljs-keyword">after insert</span><span class="hljs-punctuation">,</span> <span class="hljs-keyword">after update</span>) <span class="hljs-punctuation">{</span> <span class="hljs-comment">//NOPMD
</span>	<span class="hljs-keyword">if</span> <span class="hljs-punctuation">(</span>Quote_Custom_Sync_Config__mdt<span class="hljs-punctuation">.</span><span class="hljs-title function_ invoke__">getInstance</span><span class="hljs-params_call"><span class="hljs-punctuation">(</span><span class="hljs-string">&#x27;Use_Included_Triggers&#x27;</span><span class="hljs-punctuation">)</span></span><span class="hljs-operator">?.</span>Enabled__c <span class="hljs-operator">??</span> <span class="hljs-literal">false</span><span class="hljs-punctuation">)</span> <span class="hljs-punctuation">{</span>
		<span class="hljs-keyword">new</span> <span class="hljs-type">QuoteSyncHandler</span><span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">.</span><span class="hljs-title function_ invoke__">syncOppLines</span><span class="hljs-params_call"><span class="hljs-punctuation">(</span><span class="hljs-built_in">Trigger</span><span class="hljs-punctuation">.</span><span class="hljs-type">new</span><span class="hljs-punctuation">,</span> <span class="hljs-built_in">Trigger</span><span class="hljs-punctuation">.</span><span class="hljs-type">oldMap</span><span class="hljs-punctuation">)</span></span><span class="hljs-punctuation">;</span>
	<span class="hljs-punctuation">}</span>

	<span class="hljs-type">Integer</span> <span class="hljs-variable">notNullReturnValue</span> <span class="hljs-operator">=</span> <span class="hljs-punctuation">(</span>anInteger <span class="hljs-operator">!=</span> <span class="hljs-literal">null</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">?</span> anInteger <span class="hljs-operator">:</span> <span class="hljs-number">100</span><span class="hljs-punctuation">;</span>

	<span class="hljs-type">Integer</span> <span class="hljs-variable">notNullReturnValue</span> <span class="hljs-operator">=</span> anInteger <span class="hljs-operator">??</span> <span class="hljs-number">100</span><span class="hljs-punctuation">;</span>
<span class="hljs-punctuation">}</span>